azure-boards-common/src/main/java/com/synopsys/integration/azure/boards/common/http/AzureHttpService.java

```diff
@@ -154,22 +161,19 @@ public GenericUrl constructRequestUrl(String spec) {
             requestUrlBuilder.append("/");
         }
 
-        requestUrlBuilder.append(spec);
-
         if (!StringUtils.contains(spec, AZURE_API_VERSION_QUERY_PARAM_NAME)) {
-            char queryParamSeparator = '?';
-            if (StringUtils.contains(spec, queryParamSeparator)) {
-                queryParamSeparator = '&';
-            }
-            requestUrlBuilder.append(queryParamSeparator);
-            requestUrlBuilder.append(AZURE_API_VERSION_QUERY_PARAM_NAME);
-            requestUrlBuilder.append('=');
-            requestUrlBuilder.append(AZURE_API_VERSION);
+            spec = azureApiVersionAppender.appendApiVersion(spec, AZURE_API_VERSION);
         }
+        requestUrlBuilder.append(spec);
 
         return new GenericUrl(requestUrlBuilder.toString());
     }
 
+    public String appendApiVersion(String spec, String apiVersion) {
+        //TODO ejk - is this appropriate or should the classes that need this demand an appender?
```

Comment:
Probably the latter. Each service that needs to append a version should have an `AzureApiVersionAppender`. The version constants should probably end up in `AzureApiVersionAppender` as well.

The original code here was a misunderstanding of the Azure API (thinking that all endpoints we would use would use the same API version).

Reply:
I injected the appender and added some utility methods for appending specific versions.